src/flashbar/index.tsx

```diff
@@ -1,85 +1,178 @@
-    <Flash
-      key={item.id ?? index}
-      // eslint-disable-next-line react/forbid-component-props
-      className={clsx(isRefresh ? styles['flash-refresh'] : '')}
-      {...item}
-    />
+  useEffect(
+    function handleIsFlashbardStacked() {
+      if (enableStackingOption && items?.length > 3) {
+        setIsFlashbarStacked(true);
+      } else {
+        setIsFlashbarStacked(false);
+        setIsFlashbarStackExpanded(false);
+      }
+    },
+    [enableStackingOption, isFlashbarStacked, items]
   );
-  const renderFlashItemsWithTransitions = (items: readonly FlashbarProps.MessageDefinition[]) => {
+
+  /**
+   * If the `isFlashbarStacked` is true (which is only possible if `enableStackingOption` is true)
+   * then the first item should be rendered followed by two dummy items that visually indicate
+   * two, three, or more items exist in the stack.
+   */
+  function renderStackedItems() {
+    if (!isFlashbarStacked) {
+      return;
+    }
+
+    const stackDepth = Math.min(3, items.length);
+    const stackedItems = items.slice(0, stackDepth);
+
     return (
-      <>
-        <TransitionGroup component={null}>
-          {items &&
-            items.map((item, index) => (
-              <Transition
-                transitionChangeDelay={{ entering: TIMEOUT_FOR_ENTERING_ANIMATION }}
-                key={item.id ?? index}
-                in={true}
-              >
-                {(state: string, transitionRootElement: React.Ref<HTMLDivElement> | undefined) => (
-                  <Flash
-                    ref={transitionRootElement}
-                    key={item.id ?? index}
-                    transitionState={state}
-                    // eslint-disable-next-line react/forbid-component-props
-                    className={clsx(isRefresh ? styles['flash-refresh'] : '')}
-                    {...item}
-                  />
-                )}
-              </Transition>
+      <div className={styles.stack} style={{ [customCssProps.flashbarStackDepth]: stackDepth }}>
+        {!isFlashbarStackExpanded && (
+          <div className={clsx(styles.collapsed, isVisualRefresh && styles['visual-refresh'])}>
+            {stackedItems.map((item, index) => (
+              <div className={styles.item} style={{ [customCssProps.flashbarStackIndex]: index }} key={index}>
+                {index === 0 && renderItem(item, item.id ?? index)}
+                {index > 0 && <div className={clsx(styles.flash, styles[`flash-type-${item.type ?? 'info'}`])} />}
+              </div>
             ))}
-        </TransitionGroup>
-      </>
+          </div>
+        )}
+
+        {isFlashbarStackExpanded && (
+          <div className={styles.expanded}>{items.map((item, index) => renderItem(item, item.id ?? index))}</div>
+        )}
+
+        <button
+          aria-label={!isFlashbarStackExpanded ? 'Expand stacked notifications' : 'Collapse stacked notifications'}
```

Comment:
Let's also expose new properties for this label right away? Even for beta releases, they have to be translatable

Reply:
@just-boris Is there an example somewhere in the code I should follow?

Reply:
I think it should be `i18nStrings` property, similar to many other our components